.github/workflows/pre-commit.yaml

```diff
@@ -0,0 +1,41 @@
+name: pre-commit
+
+on:
+  push:
+    branches:
+      - main
+      - master
+      - ros2-devel
```

Comment:
Are the others besides `ros2-devel` intended to catch the names of the production branches in other repositories, i.e. this pre-commit workflow will be applied to them as well?

Reply:
Yup, I'll copy this directly into the other repos.

(I don't think additional branches should impact runtime, since this repo doesn't have branches by those names)